### DISCUSSION TITLE
Segmentation Fault - Can't connect to RDS SQL DB

### DISCUSSION THREAD

**Discussion by Gavriloski** [aws/aws-lambda-dotnet#1191] [Q&A] [answered]
I am trying to connect my AWS lambda with RDS SQL DB. 
VPC was already setup and connection is working. 
But once I try to execute any query or just open connection it fails with error :
```
RequestId: 38df0179-ecaf-45df-b6cc-a528b557b1d8 Error: Runtime exited with error: signal: segmentation fault
Runtime.ExitError
```

I tried using, EF Core, Dapper, Raw SQL connection and none of methods worked, they all give same error. 
Running locally with test tool it worked without issues on local DB. 
I think the issue is with connection string, I have no idea how it should look. 
I reused connection which is used by application, example : 

`Data Source=XXX.XXX.us-west-2.rds.amazonaws.com; Initial Catalog=XXX; Integrated Security=SSPI; MultipleActiveResultSets=True;`

I suspect issue might be with "Integrated Security=SSPI". 
Any guidance ? 

Thank you

**Comment by ashishdhingra** [comment, marked as answer]
@Gavriloski Could you please share the following:
- Share the sample code (preferably the Lambda project) to reproduce the issue.
- Are you able to connect to Amazon SQL RDS from your local machine? If yes, what does connection string looks like?
- NuGet package used along with versions (including `AWSSDK.*` packages).
- Enable client logging and send us detailed verbose logs. The easiest way to enable logging in your code is:
```C#
AWSConfigs.Logging = LoggingOptions.Console;
AWSConfigs.LogMetrics = true;
```
or, follow more detailed https://aws.amazon.com/blogs/developer/logging-with-the-aws-sdk-for-net/.
- I do not think integrated authentication works well for Amazon SQL RDS. Refer to https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/USER_ConnectToMicrosoftSQLServerInstance.html for connection options.
- Does Lambda role has proper IAM permissions to connect to SQL RDS?

Thanks,
Ashish

**Comment by Gavriloski** [comment]
@ashishdhingra 
Thanks for answer and sorry for the late replay. 
The code was very simple example, so I didn't feel like it was necessary to share. 

As suspected issue was with SSPI. Permissions and everything else was setup correctly. 
Once the SSPI was replaced with SQL user auth in connection string, connection worked without issues.
